### ISSUE TITLE
Scroll timeline by using mouse wheel without shift

### ISSUE THREAD

**Issue by Ashark** [fcpxhacks/fcpxhacks#1075]
It would be good if we could do scrolling of timeline just by scrolling mouse wheel, without need to additionally hold shift while scrolling wheel.

**Comment by latenitefilms** [comment]
I feel like this could get annoying, as I'm sure people use the scrolling mouse wheel for other tasks in Final Cut Pro (i.e. scrolling through text, etc.)?

**Comment by Ashark** [comment]
Scrolling what text? I think you mean that scrolling mouse wheel without modifier is used by default for scrolling timeline up and down to be able to see other layers that are above or bottom of primary storyline.
As for me, I use horizontal scrolling much more often. And so vertical scrolling could be assigned for something else. I feel more comfortable to scroll horizontally without shift.

**Comment by latenitefilms** [comment]
Sorry, I meant things like scrolling through a font list, or adjusting a selected Inspector parameter with the scroll wheel. I guess it might be possible to only have the scrolling timeline feature enabled if the timeline is selected to get around this?

Thoughts @randomeizer ?

**Comment by Ashark** [comment]
@latenitefilms yes, of course I was talking about scrolling timeline without shift when timeline is active. I think it should not interfere with things like scrolling in font list etc...

**Comment by randomeizer** [comment]
Or if the mouse is over the timeline?

**Comment by Ashark** [comment]
@randomeizer yes. When I am going to scroll timeline, my mouse is always over timeline.
But we should not interfere this horizontal scrolling with menu list, which appears when we right click on some clip that is on timeline.

**Comment by randomeizer** [comment]
Hmm, yeah, it gets tricky. Probably why it isn't built in...

**Comment by latenitefilms** [comment]
@randomeizer - Another option is we could take the same technqiue as the `isPlaying()` code and visually detect if the timeline is "active" via the blue border?

![screen shot 2018-03-14 at 9 19 00 pm](https://user-images.githubusercontent.com/22286696/37396571-6753dfcc-27cd-11e8-9681-45cbb60cf4c2.png)

**Comment by randomeizer** [comment]
The blue border indicates focus, but isn't always active when the timeline
is playing. If you start working on the viewer or Inspector, it won't be
focused any more, but it's still playing.

**Comment by latenitefilms** [comment]
But for the purposes of a "scroll timeline by using mouse wheel" feature, we only need to detect if it's focussed, right?

**Comment by randomeizer** [comment]
At the moment when the scroll is activated? I guess, unless there are other
'right click' menus up, etc. Maybe should check what is under the mouse
pointer at the time?

And should we expect the Timeline to be focused, or just that the mouse is
over it?

**Comment by latenitefilms** [comment]
Alejandro Fernandez Almendras [writes](https://www.facebook.com/groups/commandpost/permalink/2816458085276551/?__cft__[0]=AZX77HkkWxYWlvQ58IpbKgLhOCv3F_bOEOhGcp8QIDzoFpmPneqUUDtwIZAsBPl6BIG01ZrrtXd_JtaLxyWc9bGWJVuguOFVrC1KZqrnDhMTtfIECjnrk7fmExngCWz80A1hkjNM6Bkltj3ajaR4lrLt1Fhv9b4Fii2Waudm7EYLFtkQHBTHFi6alIBEItcgM2c&__tn__=%2CO%2CP-R):

> What an amazing addition for a Premiere user moving to FCP! Love the Zooming and the timeline moving along with the playhead. I was wondering if there's a way to use the mouse wheel to scroll left and right (without pressing the Shift key) instead of up and down. Something like this feature in Premiere.

**Comment by latenitefilms** [comment]
Richard Garside [writes](https://www.facebook.com/groups/commandpost/permalink/2816458085276551/?comment_id=2816597998595893&__cft__[0]=AZXRweVvR-1QQAKjNNHkMRh_QkPuoULH64ywLz4KBHMYL5g5uhxM9Vo6coYXLy_EmXDxTSGCsy_bIltHQcwbt7IQ0lAPYMkibGPp1WN4ExgA495_zEN9oDheMmak_YW6gFUMjUTpBl4Yl-0WdRdVg3K3&__tn__=R]-R):

> This was the first thing I thought after switching from premiere too. Shame can not at the moment

**Comment by latenitefilms** [comment]
Andrea Forgione [writes](https://www.facebook.com/groups/commandpost/posts/3037573956498295/?comment_id=3038013469787677&__cft__[0]=AZVRx898eRTU9Ecr8hvDJ1ldXuzD4Fy4dvnQcsC6nGyZPi137BQUBEoZqVNSCurGsTGKWCm7VF70BN5cHAzQoc1VIEAgVaQLvC6yE8D5BkO357QYSb0RBjmpkscd7nFzr35QcnU20O2pDiKoSbHwByCI&__tn__=R]-R):

> Thanks for your fantastic work with CommandPost! 👏
> The only thing missing is the possibility to scroll the finalcut timeline horizontally with the mouse wheel..
